numpy/ma/core.py

```diff
@@ -1615,7 +1616,7 @@ def make_mask(m, copy=False, shrink=True, dtype=MaskType):
         else:
             result = np.array(m, dtype=dtype, copy=copy)
     else:
-        result = np.array(filled(m, True), dtype=MaskType)
+        result = np.array(filled(m, True), dtype=dtype)
```

Comment:
I think this should probably be `make_mask_descr(dtype)` as in the above if statement (although I know we don't need that for this PR).

Honestly, it seems like all of these branches are just ` np.array(filled(m, True), dtype=dtype, copy=copy, subok=True)`